rectools/models/ranking/catboost_reranker.py

```diff
@@ -0,0 +1,48 @@
+import typing as tp
+
+import pandas as pd
+from catboost import CatBoostClassifier, CatBoostRanker, Pool
+
+from rectools import Columns
+
+from .candidate_ranking import Reranker
+
+
+class CatBoostReranker(Reranker):
+    """TODO: add description"""
+
+    def __init__(
+        self,
+        model: tp.Union[CatBoostClassifier, CatBoostRanker],
+        fit_kwargs: tp.Optional[tp.Dict[str, tp.Any]] = None,
+        pool_kwargs: tp.Optional[tp.Dict[str, tp.Any]] = None,
+    ):
+        super().__init__(model)
+        self.is_classifier = isinstance(model, CatBoostClassifier)
+        self.fit_kwargs = fit_kwargs
+        self.pool_kwargs = pool_kwargs
+
+    def prepare_fit_kwargs(self, candidates_with_target: pd.DataFrame) -> tp.Dict[str, tp.Any]:
```

Comment:
Let's please rename this to `prepare_training_pool` since `prepare_kwargs` sounds very lightweight and pool creation is a heavy operation (of course you need to exclude last rows from it)

Also you can do in init `self.fit_kwargs = fit_kwargs or {}` and then it will be super easy to add them during the fit call, and you don't need any methods with a strange names :)

Reply:
1. `prepare_fit_kwargs `-> `prepare_training_pool ` renamed. And now CatBoostReranker has a method `prepare_training_pool` which returns the formed `Pool` for training (instead of `tp.Dict[str, tp.Any]`)
2. Initialized `self.fit_kwargs = fit_kwargs` and during the fit call add the training_pool obtained in the previous step to fit_kwargs  `fit_kwargs = {"X": training_pool}`